gitbook-docs/urls_etc.md

```diff
@@ -69,14 +68,28 @@ If a server does not support some streaming options listed in here it must respo
 
 See [Subscription Protocol](subscription_protocol.html) for more details.
 
-##### Connection Hello
+#### Streaming Connection "Hello"
 
-Upon connection a 'hello' message is sent as follows:
+Upon accepting a connection a server MUST send a 'hello' message before any other communicaiton is sent. The format of the 'hello' message is as follows:
```

Comment:
This works when there is specific connection setup (e.g. after opening a TCP connection) but won't allow a listener to tap into a stream in progress (e.g. if they start listening to a RS422 bus or a multicast session).

Instead, suppose we add a sync message to the delta stream where the server sends all or part of the signalk tree. A minimal set would be very similar to this but by sending the tree the server can provide the context to which subsequent deltas would apply and re-synchronize clients that are new listeners or whose version of the model may have drifted.